sam/sam_test.go

```diff
@@ -842,3 +842,19 @@ func BenchmarkParseCigar(b *testing.B) {
 		}
 	}
 }
+
+func BenchmarkParseAux(b *testing.B) {
+	auxs := [][]byte{
+		[]byte("NM:i:1"),
```

Comment:
Any reason for this selection of tags? It might be worth having a variety, one benchmark for each tag type.